src/background.js

```diff
@@ -65,20 +76,30 @@ function handleWake() {
   const now = Date.now();
   log('woke at', now);
   return browser.storage.local.get().then(items => {
-    const due = Object.values(items).filter(item => item.time <= now);
+    const due = Object.entries(items).filter(entry => entry[1].time <= now);
     log('tabs due to wake', due.length);
-    return Promise.all(due.map(item =>
-      browser.tabs.create({
-        active: false,
-        url: item.url,
-        windowId: item.windowId
-      }).then(tab => browser.notifications.create(`${item.windowId}:${tab.id}`, {
-        'type': 'basic',
-        'iconUrl': browser.extension.getURL('link.png'),
-        'title': item.title,
-        'message': item.url
-      })).then(() => browser.storage.local.remove(idForItem(item)))
-    ));
+    return browser.windows.getAll({
+      windowTypes: ['normal']
+    }).then(windows => {
+      let windowIds = windows.map(window => window.id);
+      return Promise.all(due.map(([, item]) => {
+        let createProps = {
+          active: false,
+          url: item.url,
+          windowId: windowIds.indexOf(item.windowId) === -1 ? undefined : item.windowId
```

Comment:
FWIW (and it ain't "W" much), ES2015 has an [`Array.includes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) method which returns a Boolean if the array includes the specified element:

```
windowId: windowIds.includes(item.windowId) ? item.windowId : undefined
```